### DISCUSSION TITLE
Self-hosted Docker docs need to reflect upgrade

### DISCUSSION THREAD

**Discussion by udf2457** [directus/directus#16333] [Feature Requests]
The [self-hosted Docker docs](https://docs.directus.io/self-hosted/installation/docker.html) are fine, but they really need additional content to explain how to handle upgrades.

For example, the [non-Docker upgrade page](https://docs.directus.io/self-hosted/upgrades-migrations.html)  talks about updating the database schema via `npx directus database migrate:latest`.   It is not immediately apparent how such schema update would be carried out via a Docker/Kubernetes deployment.

**Comment by azrikahar** [comment]
There is the "Updating With Docker Compose" section here: https://docs.directus.io/self-hosted/installation/docker.html#updating-with-docker-compose, or perhaps you are not using Docker Compose?

Technically the last sentence in that section would still hold true for `docker run` setups (not using Docker Compose):

> Migrations will happen automatically so once the containers have started you will be on the latest version (or the version you specified).

This is because the official image defaults to the following command on start up:

https://github.com/directus/directus/blob/7899f6a2679ea5f0ffcdb29e1d722be41ca0e5c3/docker/Dockerfile#L69

in which case the [`bootstrap` command](https://docs.directus.io/reference/cli.html#bootstrap-a-project) would also migrate existing project to latest so it's as if it includes `npx directus database migrate:latest` command in it (it doesn't run the command per se, but runs the same logic!):

>  migrate it to the latest version (if it already exists and has missing migrations).

which means technically you would only need to bump up the directus image version to upgrade if you are using Docker without Docker Compose. More specifically, updating your command from:

(this is simplified, but do include volume mounts where applicable)

```
docker run \
  -p 8055:8055 \
  -e KEY=255d861b-5ea1-5996-9aa3-922530ec40b1 \
  -e SECRET=6116487b-cda1-52c2-b5b5-c8022c45e263 \
  directus/directus:9.20.3
```

to

```
docker run \
  -p 8055:8055 \
  -e KEY=255d861b-5ea1-5996-9aa3-922530ec40b1 \
  -e SECRET=6116487b-cda1-52c2-b5b5-c8022c45e263 \
  directus/directus:9.20.4
```

(if you are using `latest`, you will need to re-pull the image again to replace the `latest` tag. But it is recommended to [pin to specific versions, or tags in this case](https://docs.directus.io/self-hosted/installation/docker.html#installing-specific-versions) as a general practice)

**Comment by udf2457** [comment]
Thank you for your detailed reply @azrikahar .

Regarding your question "perhaps you are not using Docker Compose?".  Yes, my scenario would be when I refer to the Directus image in a Kubernetes manifest file.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
